build/media-suite_helper.sh

```diff
@@ -1795,3 +1829,43 @@ verify_cuda_deps() {
     fi
     enabled_any libnpp cuda-nvcc
 }
+
+check_custom_patches(){
+    local vcsFolder
+    local vcsType
+    IFS=- read vcsFolder vcsType <<< "$PWD"
+   if [ -d $LOCALBUILDDIR ] && [ -f $LOCALBUILDDIR/${vcsFolder}_extra.sh ]; then
+        source $LOCALBUILDDIR/${vcsFolder}_extra.sh
+        echo "${vcsFolder}" >> $LOCALBUILDDIR/patchedFolders
+        sort -uo $LOCALBUILDDIR/patchedFolders{,}
+    fi
+}
+
+extra_script(){
+    local stage="$1"
+    local commandname="$2"
+    if type _${stage}_${commandname} >/dev/null 2>&1; then
+        log "${stage}_${commandname}" _${stage}_${commandname}
+    fi
+}
+
+unset_extra_script(){
```

Comment:
A little tempted to do this in a loop/array. However, that would make it a little unreadable to those who might not know exactly what they are.

Reply:
Should be fine. You can do single-line comments for each or actual documentation for them if you loop them, but in the end you should minimally document at which points they're actually run.